### PR TITLE
Adding improved regexes to handle clitics c.f. #1995

### PR DESCRIPTION
**Pull request by djinn-anthrope**
The resolution to issue #1995 has been attempted. Please do take a look.

### CODE REVIEW

File: nltk/tokenize/__init__.py [hunk 1]
```diff
@@ -103,9 +103,11 @@ def sent_tokenize(text, language='english'):
 # - unicode quotes u'\u2018', u'\u2019', u'\u201c' and u'\u201d'
 
 improved_open_quote_regex = re.compile(u'([«“‘„]|[`]+)', re.U)
+improved_open_single_quote_regex = re.compile(r"(?i)(\')(?!re|ve|ll|m|t|s|d)(\w|\b", re.U)
 improved_close_quote_regex = re.compile(u'([»”’])', re.U)
 improved_punct_regex = re.compile(r'([^\.])(\.)([\]\)}>"\'' u'»”’ ' r']*)\s*$', re.U)
 _treebank_word_tokenizer.STARTING_QUOTES.insert(0, (improved_open_quote_regex, r' \1 '))
+_treebank_word_tokenizer.STARTING_QUOTS.append((impoved_open_single_quote_regex, r'\1 \2'))
```
Comment (alvations) [inline nltk/tokenize/__init__.py]: There's a typo here. It should be:

```python
_treebank_word_tokenizer.STARTING_QUOTES.append((improved_open_single_quote_regex, r'\1 \2'))
```

**Review by alvations — CHANGES_REQUESTED** [review]
After fixing the typo, it should be as suggested in #1995 , then it would be okay to merge.

**Review by alvations — DISMISSED** [review]

**Comment by alvations** [comment]
Still needs some tests for this. The existing doctest for the tokenizer might break because of the change. Maybe we should add an argument to allow users to choose whether they want to handle clitics.

**Comment by alvations** [comment]
[CI: retest]

**Review by alvations — APPROVED** [review]
LGTM too!

**Comment by stevenbird** [comment]
Thanks @AlokDebnath, @alvations

**Pull request merged**